### PR TITLE
add mock server and run_server, run_client scripts

### PR DESCRIPTION
**Pull request by botka1998**
Added ```MockServer``` to mock the robot controller, this server can receive instructions from ```abb.Robot``` client and print out the received instruction. 
### Usage
Implement the robot instructions as ```MockServer``` methods and add the mapping to ```MockServer.function_dict```
Reminder, instructions are in the format ```XX x1 x2 x3 ... xn #``` 
where
- XX is the function id defined by the server-client contract e.g. "00", "01"...
- x1, x2, x3 ... xn are function input parameters separated by 1 whitespace 
- \# indicates the end of an instruction string without which the instruction is invalid

### CODE REVIEW

File: controllers/mock/models.py
Comment (milistu) [inline controllers/mock/models.py:1]: We need to add more Orientation representations. For now, this is okay.

File: controllers/mock/server.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        self.connection.bind(("", self.port))
+        self.function_dict = {
+            "01": self.move_tcp,
+            "02": self.set_joints,
+            "04": self.get_joints,
+            "06": self.set_tool,
+        }
+
+    def run(self):
```
Comment (milistu) [inline controllers/mock/server.py]:
```suggestion
    def run(self) -> None:
```

File: controllers/mock/server.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        if message[-1] != "#":
+            print("[ERROR] Invalid instruction, wrong format")
+            return None, []
+
+        instruction = message[:-1].split()
+        action, parameters = instruction[0], instruction[1:]
+        return action, parameters
+
+    def get_function(self, action) -> Callable[[any], None] | None:
```
Comment (milistu) [inline controllers/mock/server.py]:
```suggestion
    def get_function(self, action:str) -> Callable[[any], None] | None:
```

File: controllers/mock/server.py (hunk 1 above)
Comment (milistu) [reply controllers/mock/server.py]: Also, add missing Type Hints to other methods.

File: controllers/mock/server.py
Comment (milistu) [inline controllers/mock/server.py:1]: We need to replace print statements with logger. I am creating a task and related issue for this.

File: functions/run_client.py
Comment (milistu) [inline functions/run_client.py:1]: Please add real tests here.

**Review by milistu — CHANGES_REQUESTED** [review]
Please check the suggestions and fix them 🔨

File: controllers/mock/server.py
Comment (botka1998) [reply controllers/mock/server.py:1]: Please link the issue here <3

Comment (milistu) [reply controllers/mock/server.py:1]: Done ✅

**Review by milistu — APPROVED** [review]
Everything looks good to me 👍

**Pull request merged**